pvanalytics/features/clipping.py

```diff
@@ -223,3 +226,110 @@ def threshold(ac_power, slope_max=0.0035, power_min=0.75,
         freq=freq
     )
     return ac_power >= threshold
+
+
+def _estimate_freq_minutes(index):
+    # Estimate the timestamp frequency in minutes.
+    #
+    # Parameters
+    # ----------
+    # index : DatetimeIndex
+    #
+    # Returns
+    # -------
+    # float
+    #     Timestamp spacing in minutes.
+    freq = pd.infer_freq(index)
+    if freq:
+        return pd.to_timedelta(frequencies.to_offset(freq)).seconds / 60
+    return index.to_series().diff().astype('timedelta64[m]').mode()[0]
```

Comment:
I'm not sure whether we want to keep this function or just have users pass the frequency as a param when it cannot be inferred from the index. Passing an extra param would be more consistent with how we have done things in other functions ported from PVFleets